### DISCUSSION TITLE
Profile files loaded when EnvironmentVariableCredentialsProvider is used

### DISCUSSION THREAD

**Discussion by c11v** [awslabs/aws-sdk-rust#758] [Q&A] [answered]
I'm running a Rust lambda and looking to configure the credentials to use `EnvironmentVariableCredentialsProvider`. 

```rust
let config = aws_config::from_env()
        .region(Region::new(env_var("AWS_REGION")))
        .credentials_provider(EnvironmentVariableCredentialsProvider::new())
        .load()
        .await;

fn env_var(name: &str) -> String {
    match env::var(name) {
        Ok(v) => v,
        Err(e) => panic!("${} is not set ({})", name, e)
    }
}
```
This works in AWS lambda, however, I see that before loading the credentials from environment it will attempt to load the profile files: 

```
....138408Z DEBUG load_config_file{file=Default(Config)}: config file not found path=~/.aws/config
....138444Z DEBUG load_config_file{file=Default(Config)}: config file loaded path=Some("~/.aws/config") size=0
....138459Z DEBUG load_config_file{file=Default(Credentials)}: config file not found path=~/.aws/credentials
....138462Z DEBUG load_config_file{file=Default(Credentials)}: config file loaded path=Some("~/.aws/credentials") size=0
....171764Z TRACE Loading config from env
```

That's `33 ms` wasted loading the profile file when I'm specifically asking to use `EnvironmentVariableCredentialsProvider`. 

My next attempt was to remove the profile files:

```rust
let config = aws_config::from_env()
        .region(Region::new(env_var("AWS_REGION")))
        .credentials_provider(EnvironmentVariableCredentialsProvider::new())
        .profile_files(ProfileFiles::builder()
            .include_default_config_file(false)
            .include_default_credentials_file(false)
            .build()
        )
        .load()
        .await;
```

However, it seems at least one profile file is required:

```
thread 'main' panicked at 'At least one profile file must be included in the `ProfileFiles` file set.', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/aws-config-0.54.1/src/profile/profile_file.rs:199:13
```

I'm wondering why the profile files are always loaded even when `EnvironmentVariableCredentialsProvider` is specified. I looked through the implementation and I could not find a way to stop this behavior.

**Comment by rcoh** [comment, marked as answer]
I doubt that the 33ms is _actually_ from loading the profile file. You can hardcode an empty profile file if you want to test this—https://docs.rs/aws-config/latest/aws_config/profile/profile_file/struct.Builder.html#method.with_contents. 

Let me know what you find!

You might also try https://github.com/rcoh/tracing-texray to see what's going on
